server/src/main/java/io/crate/analyze/where/WhereClauseAnalyzer.java

```diff
@@ -143,20 +143,21 @@ public static PartitionResult resolvePartitions(Symbol query,
             }
         }
 
+        if (queryPartitionMap.isEmpty()) {
+            return new PartitionResult(Literal.BOOLEAN_FALSE, Collections.emptyList());
+        }
         if (queryPartitionMap.size() == 1) {
             Map.Entry<Symbol, List<Literal<?>>> entry = Iterables.getOnlyElement(queryPartitionMap.entrySet());
             return new PartitionResult(
                 entry.getKey(), Lists.map(entry.getValue(), literal -> nullOrString(literal.value())));
-        } else if (queryPartitionMap.size() > 0) {
```

Comment:
I think this changes the execution path taken for `queryPartitionMap.size() == 0`

Reply:
if it's empty, is moved first, then if == 1 and else if it's > 1, do you still see an issue, should I revert?

Reply:
Oh sorry didn't see that. Please proceed.